server/src/main/java/org/elasticsearch/action/support/replication/TransportWriteAction.java

```diff
@@ -94,8 +95,7 @@ protected TransportWriteAction(
             request,
             replicaRequest,
             EsExecutors.DIRECT_EXECUTOR_SERVICE,
-            true,
-            forceExecutionOnPrimary
+            ArrayUtils.including(flags, ActionFlags.SyncGlobalCheckpointAfterOperation)
```

Comment:
In this case, all `TransportWriteActions` `SyncGlobalCheckpointAfterOperation` so we add that if it isn't present.

A shift to a set of flags means subclasses need to decide if they pass through the flags, or modify them in some way, or present their own flags or booleans representing a restricted subset of the superclass' flags.